src/mesh/voxel_cloud.rs

```diff
@@ -448,6 +448,165 @@ impl ScalarField {
+                    // the neighboring voxels
+                    let corners_absolute_coordinates = [
+                        Point3::new(x, y + 1, z),
+                        Point3::new(x + 1, y + 1, z),
+                        Point3::new(x + 1, y, z),
+                        Point3::new(x, y, z),
+                        Point3::new(x, y + 1, z + 1),
+                        Point3::new(x + 1, y + 1, z + 1),
+                        Point3::new(x + 1, y, z + 1),
+                        Point3::new(x, y, z + 1),
+                    ];
+                    let corners_cartesian_coordinates: Vec<_> = corners_absolute_coordinates
+                        .iter()
+                        .map(|a| absolute_voxel_to_cartesian_coordinate(a, &self.voxel_dimensions))
+                        .collect();
+
+                    // These points will become vertices of mesh patches forming
+                    // the outer envelope of the scalar field.
+                    #[rustfmt::skip]
+                    let edge_midpoints = vec![
+                        Point3::from(corners_cartesian_coordinates[0].coords.lerp(&corners_cartesian_coordinates[1].coords, 0.5)), //e0
+                        Point3::from(corners_cartesian_coordinates[1].coords.lerp(&corners_cartesian_coordinates[2].coords, 0.5)), //e1
+                        Point3::from(corners_cartesian_coordinates[2].coords.lerp(&corners_cartesian_coordinates[3].coords, 0.5)), //e2
+                        Point3::from(corners_cartesian_coordinates[3].coords.lerp(&corners_cartesian_coordinates[0].coords, 0.5)), //e3
+                        Point3::from(corners_cartesian_coordinates[4].coords.lerp(&corners_cartesian_coordinates[5].coords, 0.5)), //e4
+                        Point3::from(corners_cartesian_coordinates[5].coords.lerp(&corners_cartesian_coordinates[6].coords, 0.5)), //e5
+                        Point3::from(corners_cartesian_coordinates[6].coords.lerp(&corners_cartesian_coordinates[7].coords, 0.5)), //e6
+                        Point3::from(corners_cartesian_coordinates[7].coords.lerp(&corners_cartesian_coordinates[4].coords, 0.5)), //e7
+                        Point3::from(corners_cartesian_coordinates[0].coords.lerp(&corners_cartesian_coordinates[4].coords, 0.5)), //e8
+                        Point3::from(corners_cartesian_coordinates[1].coords.lerp(&corners_cartesian_coordinates[5].coords, 0.5)), //e9
+                        Point3::from(corners_cartesian_coordinates[2].coords.lerp(&corners_cartesian_coordinates[6].coords, 0.5)), //e10
+                        Point3::from(corners_cartesian_coordinates[3].coords.lerp(&corners_cartesian_coordinates[7].coords, 0.5)), //e11
+                    ];
+
+                    // Boolean vector marking each of the voxels either volume
+                    // (`true`) or void(`false`)
+                    let corners_inside_volume_pattern: Vec<_> = corners_absolute_coordinates
```

Comment:
Use an an array. No need to heap-allocate.

Reply:
ArrayVec, since it's a `bool` structure

Reply:
I don't follow your argument.

Reply:
had to use `ArrayVec` because of this:

```
error[E0277]: a collection of type `[bool; 8]` cannot be built from an iterator over elements of type `bool`
    --> src\mesh\voxel_cloud.rs:1671:30
     |
1671 | ...                   .collect();
     |                        ^^^^^^^ a collection of type `[bool; 8]` cannot be built from `std::iter::Iterator<Item=bool>`      
     |
     = help: the trait `std::iter::FromIterator<bool>` is not implemented for `[bool; 8]`
```

Reply:
See my comment down below (github didn't allow me to reply here)